toolchain/check/type_structure.cpp

```diff
@@ -48,64 +51,67 @@ auto TypeStructure::IsCompatibleWith(const TypeStructure& other) const -> bool {
     // From here we know one side is a Symbolic and the other is not. We can
     // match the Symbolic against either a single Concrete or a larger bracketed
     // set of Concrete structural elements.
```

Comment:
Optional: instead of checking for one of the two operands being symbolic here, maybe integrate this into the `if`s on line 99-107.